File: Backend/AccommodationService/Model/Accommodation.cs
using System.Security.Cryptography.X509Certificates;
using System.Text.RegularExpressions;
using AccommodationService.Enums;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace AccommodationService.Model
{
    public class Accommodation
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? Id { get; set; }
        public String Name { get; set; }
        public String Location { get; set; }
        public List<String> Facilities { get; set; }
        public List<String> Photos { get; set; }
        public int MinCapacity { get; set; }
        public int MaxCapacity { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public double Price { get; set; }
        public ReservationType ReservationType { get; set; }
        public double Grade { get; set; }
        public string HostId { get; set; }

        


    }
}

File: Frontend/src/app/model/accommodation.ts
import { Reservation } from "./reservation";

export class Accommodation {

    id: string = "";
    name : string = '';
    location : string = '';
    facilities: string[] = [];
    photos: string[] = [];
    minCapacity: number = 0;
    maxCapacity: number = 0;
    startDate: Date = new Date();
    endDate: Date = new Date();
    price: number = 0;
    reservationType: Reservation = Reservation.Autoautomatically;
    grade: number = 0;
  
    constructor(obj?: any) {
      if (obj) {
        this.id = obj.id;
        this.name = obj.name;
        this.location = obj.location;
        this.facilities = obj.facilities;
        this.photos = obj.photos;
        this.minCapacity = obj.minCapacity;
        this.maxCapacity = obj.maxCapacity;
        this.startDate = obj.startDate;
        this.endDate = obj.endDate;
        this.price = obj.price;
        this.reservationType = obj.reservationType;
        this.grade = obj.grade;
      }
    }
  }
  